subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompileWithAnnotationProcessingBuildOperationType.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.Map;
+
+public class CompileWithAnnotationProcessingBuildOperationType implements BuildOperationType<CompileWithAnnotationProcessingBuildOperationType.Details, CompileWithAnnotationProcessingBuildOperationType.Result> {
```

Comment:
Should have `@UsedByScanPlugin`

Reply:
That's certainly the intention :)

Reply:
Added in 7f3b681da26.

Reply:
Let’s remove this until it is actually used.

This annotation is there to signal that there may be non obvious impacts of changing this API due to build scans. That is not the case here.

Reply:
Alright, I'll remove it in a subsequent PR.